### DISCUSSION TITLE
Serial Connection Via Luban to SM 2.0 Not Connecting

### DISCUSSION THREAD

**Discussion by Mississippi12** [Snapmaker/Luban#2466] [Q&A] [answered]
Hello,

I'm having difficulty connecting Luban Workspace to my A350. When I select "Serial," the port "/dev/ttyUSB0" appears in the drop-down menu, but when I click "Connect," the button greys out and and a spinning circle appears in it. It never connects even after waiting for some time.

Printer firmware is updated to V1.18.2, Luban is V4.10.2, OS is Zorin 16.3 Lite (based on Ubuntu 22.04).

It appears that CH340 drivers are installed and functioning as far as I can tell.

Is there something that I missed when installing Luban?

**Comment by apartmedia** [comment]
A bit off topic but establishing a connection feels getting a bit worse with every update of Luban (or hub FW?).
My SM 2.0 is not able to auto connect to WiFi in a way, that Luban can communicate to the machine after it is powered on.
Although the printer shows connected and internet access true in the machine settings, the Luban computer can not reach it (no ping possible). I always need to disable/enable WiFi several times at the printer before it‘s connected and reachable in the network.
May be there’s something buggy in the SM FW that auto connect WiFi and also serial is not as stable as expected.

**Comment by Mississippi12** [comment]
New development in my issue. I have read that several people like using OctoPrint to connect to their Snapmakers. I decided to try installing it on my computer and attempt a connection with it. However, it also failed to connect. This leads me to wonder if it is a printer side issue, where the printer is not excepting or responding to requests to open a connection.

I started my attempts to connect with a controller firmware version of 1.15.xx, if memory serves. After failing to connect, I updated to 1.18.2 as I mention above. So, did something in the firmware change in the versions prior to 1.15? Do my findings with OctoPrint suggest issues with the printer or my computers hardware?

Also, would there be useful information in the logs from both the printer and Luban? I will be checking them anyway.

Any incite is much appreciated.

**Comment by Mississippi12** [comment, marked as answer]
I have resolved the connection issues.

For those who may be having a similar issue and for posterity, I'll detail my findings here:

I exported the logs from both the printer and Luban, and found that Luban was throwing an error something along the lines of:

       Access denied: could not access /dev/ttyUSB0

I solved this by adding my user account to the "dialout" group (which the /dev/ttyUSB0 port is under) with:

      sudo usermod -a -G dialout $USER

I also read that a package called "modemmanager" may also cause issues with access to the "/dev/ttyUSB0" port. So, I also removed that:

      sudo apt-get remove modemmanager

This package may be needed by some people, so I would only remove it if necessary. (my computer is offline and does not use any networking)

I hope this helps others facing similar problems. I also hope that Snapmaker adds an error message to alert users regarding this kind of permissions issue.

Cheers!
